[amir-dha] iP

amir-dha opened the pull request:
# Mochi
> **_Bullying you into success, one task at a time._**
 
## Mochi, who?

###  _Say hi to Mochi, who ironically, is not sweet at all and **WILL** badger you to be on task. It's giving:_
- :family_man_woman_girl_boy: Kiasu Parent Vibes (KPV)
- :+1: Very gwenchana to use
- :computer: An online chingu, no cap

## How to get Started?

### _Well, welcome to your first task:_
- [ ] Download the latest version [here](https://github.com/amir-dha/ip/releases/download/A-Jar/mochi.jar).
- [ ] Run the JAR file.
- [ ] Start adding things you are totally "going" to do.

## Features

1. :calendar: Task Organization, because we should.
2. :hourglass: Deadlines & Events tracking, because we hope.
3. :fire: Persistent file-based storage, because we care.
4. :microphone: A lovely yapper, because we can.

### Now, if you are Java bro, you can use it to practise Java leh. Here, got main method:
```
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

amir-dha committed:
This is Level 0.

amir-dha committed:
This is Level 1.

amir-dha committed:
This is Level 2.

amir-dha committed:
This is Level 3.

amir-dha committed:
This is Level 4.

amir-dha committed:
This is Level 5.

amir-dha committed:
This is Level 6.

amir-dha committed:
Corrected all the tags.

amir-dha committed:
Corrected all the tags.

amir-dha committed:
Done testing.

amir-dha committed:
Implemented Level-7: Saving and Loading tasks

amir-dha committed:
Merge Level-7 with a merge commit

amir-dha committed:
Implemented Level-8: Date and Time Support

amir-dha committed:
Merge Level-8 with a merge commit

amir-dha committed:
Implemented MoreOOP: Extracted various classes

amir-dha committed:
Merge A-MoreOOP with a merge commit

amir-dha committed:
Implemented Packages: Separated the files into different packages.

amir-dha committed:
Merge branch 'add-gradle-support'
Merging add-gradle-support branch

amir-dha committed:
Set up Gradle build automation

amir-dha committed:
Added JUnit tests for Mochi and TaskList

amir-dha committed:
Merge A-JUnit  with a merge commit

amir-dha committed:
Added JavaDoc comments to all relevant classes.

amir-dha committed:
Making some tweaks for Coding Standard.

amir-dha committed:
Merge branch-A-JavaDoc.

amir-dha committed:
Merge branch-A-CodingStandard.

amir-dha committed:
Implemented Level-9: Find Command.

amir-dha committed:
Resolved merge conflicts by keeping master's Gradle files

NightlyAffair commented on src/main/java/mochi/command/Command.java:
```diff
@@ -0,0 +1,158 @@
+import mochi.task.Todo;
+import mochi.task.Deadline;
+import mochi.task.Event;
+import mochi.task.TaskList;
```
I think in the case of more than 4 imports, it is acceptable to use wildcard imports. Noticed this in other files as well.

`import mochi.task.*`

NightlyAffair commented on src/main/java/mochi/command/Parser.java:
Personally love the personality in the interactions with users

NightlyAffair commented on src/main/java/mochi/command/Parser.java:
```diff
@@ -0,0 +1,78 @@
+     * @throws MochiException MochiException if the input is invalid.
+     */
+    public static Command parse(String input) throws MochiException {
+        String[] pts = input.split(" ", 2);
```
Perhaps pts could be named better maybe userCommand?

NightlyAffair commented on src/main/java/mochi/storage/Storage.java:
```diff
@@ -0,0 +1,123 @@
+    private Task parseTask(String line) {
+        try {
+            String[] parts = line.split(" \\| ");
+            String type = parts[0];
```
Could use ENUMs for the type of user command as it is a constant allowing for easier naming

NightlyAffair reviewed (commented):
I think most variables are well named. However i feel that there is a tendency for shorter names where longer names would aid in understanding and readability.

Kannan171 commented on src/main/java/mochi/Mochi.java:
```diff
@@ -0,0 +1,76 @@
+
+        while (!isBye) {
+            try {
+                String instruction = scanner.nextLine().trim();
+                Command command = Parser.parse(instruction);
+                command.exec(tasks, ui, storage);
+                isBye = command.isBye();
```
LGTM, well abstracted, code is very neat

Kannan171 commented on src/main/java/mochi/task/Task.java:
```diff
@@ -0,0 +1,55 @@
+/**
+ * Represents a general task with a description and completion status.
+ */
+public abstract class Task {
```
I like your OOP approach to the tasks

Kannan171 commented on src/main/java/mochi/command/Command.java:
```diff
@@ -0,0 +1,158 @@
+/**
+ * Represents an abstract command that can be executed on the task list.
+ */
+public abstract class Command {
```
Instead of having so many different classes, perhaps it may be better to have them as separate methods instead?

Kannan171 reviewed (commented):
LGTM

amir-dha committed:
Add Checkstyle.

amir-dha committed:
Add GUI to chatbot

amir-dha committed:
Merge branch-Level-10

amir-dha committed:
Refactor BCD extension and improve code quality

The codebase for the Mochi chatbot required improvements in the
BCD extension implementation and general code quality.

Currently, Ui.java prints messages directly using System.out, which is
not suitable for a GUI-based chatbot. Additionally, unused CLI-based
methods remain in the code, adding unnecessary complexity.

Furthermore, MainWindow.java does not properly initialize the welcome
message for the GUI, requiring manual input to trigger a response.
Mochi.java handles error messages inconsistently, printing some errors
while returning others.

Let's,
* Refactor Ui.java to return formatted messages instead of printing
* Update Mochi.java to use Ui for better error handling
* Modify MainWindow.java to display the welcome message on startup
* Ensure commands properly interact with the GUI through DialogBox
* Remove unused CLI-based methods to improve maintainability

These changes ensure that the chatbot functions seamlessly within the
GUI, improving modularity and readability while keeping only relevant
code.

amir-dha committed:
Refactor BCD extension and improve code quality

The codebase for the Mochi chatbot required improvements in the
BCD extension implementation and general code quality.

Currently, Ui.java prints messages directly using System.out, which is
not suitable for a GUI-based chatbot. Additionally, unused CLI-based
methods remain in the code, adding unnecessary complexity.

Furthermore, MainWindow.java does not properly initialize the welcome
message for the GUI, requiring manual input to trigger a response.
Mochi.java handles error messages inconsistently, printing some errors
while returning others.

Let's,
* Refactor Ui.java to return formatted messages instead of printing
* Update Mochi.java to use Ui for better error handling
* Modify MainWindow.java to display the welcome message on startup
* Ensure commands properly interact with the GUI through DialogBox
* Remove unused CLI-based methods to improve maintainability

These changes ensure that the chatbot functions seamlessly within the
GUI, improving modularity and readability while keeping only relevant
code.

amir-dha committed:
Refactor BCD extension and improve code quality

amir-dha committed:
Refactor BCD extension and improve code quality

The codebase for the Mochi chatbot required improvements in the
BCD extension implementation and general code quality.

Currently, Ui.java prints messages directly using System.out, which is
not suitable for a GUI-based chatbot. Additionally, unused CLI-based
methods remain in the code, adding unnecessary complexity.

Furthermore, MainWindow.java does not properly initialize the welcome
message for the GUI, requiring manual input to trigger a response.
Mochi.java handles error messages inconsistently, printing some errors
while returning others.

Let's,
* Refactor Ui.java to return formatted messages instead of printing
* Update Mochi.java to use Ui for better error handling
* Modify MainWindow.java to display the welcome message on startup
* Ensure commands properly interact with the GUI through DialogBox
* Remove unused CLI-based methods to improve maintainability

These changes ensure that the chatbot functions seamlessly within the
GUI, improving modularity and readability while keeping only relevant
code.

amir-dha committed:
Refactor code for A-Stream, A-Personality, and A-MoreErrorHandling

amir-dha committed:
Improve GUI: Added stretchable header, pastel theme, and aligned user text

amir-dha committed:
Implement Java assertions for key assumptions in the code

amir-dha committed:
Main and TaskListTest: Add Java assertions for key assumptions

Mochi currently lacks runtime checks to validate critical assumptions
in key components like initialization and task handling.

This lack of validation could allow unexpected states to propagate,
leading to hard-to-debug errors.

Let's,
- Add Java assertions in `Main.java` to ensure proper initialization.
- Include assertions in `TaskListTest.java` to validate task additions.
- Ensure Storage correctly handles file loading and saving.

Using Java assertions helps enforce assumptions during development
while allowing them to be disabled in production for performance.

The codebase for the Mochi chatbot required improvements in the
BCD extension implementation and general code quality.

Currently, Ui.java prints messages directly using System.out, which is
not suitable for a GUI-based chatbot. Additionally, unused CLI-based
methods remain in the code, adding unnecessary complexity.

Furthermore, MainWindow.java does not properly initialize the welcome
message for the GUI, requiring manual input to trigger a response.
Mochi.java handles error messages inconsistently, printing some errors
while returning others.

Let's,
* Refactor Ui.java to return formatted messages instead of printing
* Update Mochi.java to use Ui for better error handling
* Modify MainWindow.java to display the welcome message on startup
* Ensure commands properly interact with the GUI through DialogBox
* Remove unused CLI-based methods to improve maintainability

These changes ensure that the chatbot functions seamlessly within the
GUI, improving modularity and readability while keeping only relevant
code.

amir-dha committed:
Merge pull request #2 from amir-dha/branch-A-Assertions

Add Java Assertions for Key Assumptions (A-Assertions)

amir-dha committed:
Resolved Gradle-related conflicts by regenerating build files

amir-dha committed:
Add Ui.png to the docs folder

amir-dha committed:
Add detailed User Guide with setup instructions and commands

amir-dha committed:
Delete extra file UserGuide

amir-dha committed:
Update README.md

amir-dha committed:
Update README.md

amir-dha committed:
Delete outdated files

amir-dha committed:
Update README.md

amir-dha committed:
Enhance GUI exit: Delay window close after 'bye' command

amir-dha committed:
Fixed event, deadline, and sort command validation and added proper error handling

amir-dha committed:
Delete Outdated Files

amir-dha committed:
Fix issues with exit

amir-dha committed:
Remove data file

amir-dha committed:
Move old mochi.jar file to jar_files_first_version

amir-dha committed:
TaskListTest.java: Remove blank lines

amir-dha committed:
MochiTest.java: Remove blank lines

amir-dha committed:
Command.java: Remove blank lines

amir-dha committed:
Parser.java: Remove blank lines
